src/components/Tool/FixCommitmentInfo/Contact.tsx

```diff
@@ -245,30 +250,28 @@ const Contact: React.FC<Props> = ({
     <Grid container className={classes.container}>
       <Formik
         initialValues={{
-          statusValue: suggestedStatus || statusValue,
+          status: suggestedStatus || status,
           pledgeCurrency: amountCurrency,
           pledgeAmount: suggestedAmount || amount,
-          pledgeFrequency: suggestedFrequency || frequencyValue,
+          pledgeFrequency:
+            frequencyOptions?.find(
+              (frequency) => frequency?.key === suggestedFrequency,
+            )?.id || frequencyValue,
```

Comment:
Test to see how often this is  called, but it might be better to save the result to a useMemo above. But test it out.